src/java.base/share/classes/sun/util/locale/BaseLocale.java

```diff
@@ -165,8 +162,21 @@ public static BaseLocale getInstance(String language, String script,
             language = convertOldISOCodes(language);
         }
 
-        Key key = new Key(language, script, region, variant, false);
-        return Cache.CACHE.get(key);
+        // Obtain the "normalized" BaseLocale, using un-normalized
+        // BaseLocale as the key. The returned "normalized" instance
+        // can subsequently be used by the Locale instance which
+        // guarantees the locale components are properly cased/interned.
+        LOCK.lock();
+        try {
+            return CACHE.computeIfAbsent(new BaseLocale(language, script, region, variant),
+                    (b) -> new BaseLocale(
+                            LocaleUtils.toLowerString(b.getLanguage()).intern(),
+                            LocaleUtils.toTitleString(b.getScript()).intern(),
+                            LocaleUtils.toUpperString(b.getRegion()).intern(),
```

Comment:
don't lines 147 and 148 above already guarantee the language and region have correct case?

Reply:
147,148 are in `getInstance()` and the BaseLocale key here is created with a constructor, which simply copies the passed arguments. So normalization is needed here

Reply:
perhaps I am missing something, but aren't `language` and `region` being reassigned to the case adjusted values? So the values used to create the "key" BaseLocale instance reflect that case adjustment? (language is again potentially adjusted at line 162 to account for old iso codes, but that never results in the wrong case.)

```
        language = LocaleUtils.toLowerString(language);
        region = LocaleUtils.toUpperString(region);
```